src/python/pants/core/goals/test.py

```diff
@@ -115,10 +126,14 @@ class EnrichedTestResult(TestResult, EngineAwareReturnType):
 
     output_setting: ShowOutput = ShowOutput.ALL
 
-    def artifacts(self) -> Optional[Dict[str, Snapshot]]:
-        if not self.xml_results:
-            return None
-        return {"xml_results": self.xml_results}
+    def artifacts(self) -> Optional[Dict[str, Union[FileDigest, Snapshot]]]:
```

Comment:
Is it worth adding a type alias for `Dict[str, Union[FileDigest, Snapshot]]` ? It's a bit of a mouthful.
```
WorkunitArtifacts = Dict[str, Union[FileDigest, Snapshot]]
```